src/dev/update_check.ts

```diff
@@ -0,0 +1,126 @@
+import { colors, dirname, join } from "./deps.ts";
+
+interface CheckFile {
+  last_checked: string;
+  latest_version: string;
+  current_version: string;
+}
+
+function getHomeDir(): string | null {
+  switch (Deno.build.os) {
+    case "linux": {
+      const xdg = Deno.env.get("XDG_CACHE_HOME");
+      if (xdg) return xdg;
+
+      const home = Deno.env.get("HOME");
+      if (home) return `${home}/.cache`;
+      break;
+    }
+
+    case "darwin": {
+      const home = Deno.env.get("HOME");
+      if (home) return `${home}/Library/Caches`;
+      break;
+    }
+
+    case "windows":
+      return Deno.env.get("LOCALAPPDATA") ?? null;
+  }
+
+  return null;
+}
+
+function getFreshCacheDir(): string | null {
+  const home = getHomeDir();
+  if (home) return join(home, "fresh");
+  return null;
+}
+
+async function fetchLatestVersion() {
+  const res = await fetch("https://dl.deno.land/fresh/release-latest.txt");
+  if (res.ok) {
+    return (await res.text()).trim().replace(/^v/, "");
+  }
+
+  throw new Error(`Could not fetch latest version.`);
+}
+
+export async function updateCheck(
+  interval: number,
+  getCacheDir = getFreshCacheDir,
+  getLatestVersion = fetchLatestVersion,
+) {
+  // Skip update checks on CI or Deno Deploy
+  const FRESH_NO_UPDATE_CHECK = Deno.env.get("FRESH_NO_UPDATE_CHECK");
+  if (
+    Deno.env.get("CI") === "true" || FRESH_NO_UPDATE_CHECK === "true" ||
+    FRESH_NO_UPDATE_CHECK === "1" ||
+    Deno.env.get("DENO_DEPLOYMENT_ID")
+  ) {
+    return;
+  }
+
+  // Abort if we couldn't find a deno_dir
```

Comment:
This comment looks outdated?
```suggestion
  // Abort if we couldn't find fresh cache dir
```